src/util.js

```diff
@@ -2,32 +2,46 @@ const through = require('through2').obj
 const pump = require('pump')
 const pify = require('pify')
 
+const DEFAULT_TRANSFORM_OPTIONS = {
+  highWaterMark: 16,
+  objectMode: true
+}
+
 class ModuleGroup {
-  constructor ({ label, parentGroup }) {
+  constructor ({ label, parentGroup, ...streamOptions }) {
     this.label = label
     this.parentGroup = parentGroup
-    this.stream = through()
+    this.stream = through({
+      ...DEFAULT_TRANSFORM_OPTIONS,
+      ...streamOptions
+    })
   }
 }
 
 module.exports = { createForEachStream, getStreamResults, ModuleGroup }
 
 const noop = () => {}
 
-function createForEachStream ({ onEach = noop, onEnd = noop }) {
-  return through(
-    (entry, _, cb) => { onEach(entry); cb() },
-    (cb) => { onEnd(); cb() }
+function createForEachStream ({ onEach = noop, onEnd = noop, ...streamOptions }) {
+  return through({
+    ...DEFAULT_TRANSFORM_OPTIONS,
+    ...streamOptions
+  },
+  (entry, _, cb) => { onEach(entry); cb() },
+  (cb) => { onEnd(); cb() }
   )
 }
 
-async function getStreamResults (stream) {
+async function getStreamResults (stream, streamOptions) {
```

Comment:
can we default `streamOptions` to `DEFAULT_TRANSFORM_OPTIONS ` to make it optional ?

Reply:
So in this function, it's only used on one place, passed into `createForEachStream`. If this is omitted, `streamOptions` will be `undefined` there, and already filled in by the `...DEFAULT_TRANSFORM_OPTIONS` spread on line 27. So it should already be optional.

Does that make sense? :)